common/bitcointree/musig2.go

```diff
@@ -507,3 +523,71 @@ type treePrevOutFetcher struct {
 func (f *treePrevOutFetcher) FetchPrevOutput(wire.OutPoint) *wire.TxOut {
 	return f.prevout
 }
+
+type writable interface {
+	Encode(w io.Writer) error
+}
+
+type readable interface {
+	Decode(r io.Reader) error
+}
+
+// encodeMatrix encode a matrix of serializable objects into a byte stream
+func encodeMatrix[T writable](matrix [][]T) ([]byte, error) {
```

Comment:
```suggestion
func encodeMatrix(matrix [][]writable) ([]byte, error) {
```

Reply:
no need for generics?

Reply:
We need it because the interface is not the argument itself. Instead, the argument is a slice matrix of the interface. I won't speculate why Golang does not accept this but if I make the change you suggest, I need to cast the matrix passed as arguments: "`TreeNonces` is not a `[][]writable`"